src/main/java/seedu/address/model/moduleclass/UniqueModuleClassList.java

```diff
@@ -117,6 +120,21 @@ public void remove(ModuleClass toRemove) throws NullPointerException, ModuleClas
         }
     }
 
+    /**
+     * Removes the specified {@code UUID} from all {@ModuleClass}es in the class list.
+     */
+    public void removeUuid(UUID uuidToRemove) {
+        requireNonNull(uuidToRemove);
+
+        internalList.setAll(internalList.stream()
+                .map(moduleClass -> {
+                    Set<UUID> modifiedUuids = new HashSet<>(moduleClass.getStudentUuids());
```

Comment:
Just to clarify, the reason why you create a new hashset is because the set returned by ```getStudentUuids()``` is immutable?

Reply:
Yup